### PR TITLE
Linalg linalg

### PR DESCRIPTION
**Pull request by alewis**
Adds linalg/linalg.py, which contains decompositions and so on.

### CODE REVIEW

File: tensornetwork/linalg/tests/test_linalg.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import tensornetwork
+from tensornetwork.linalg import linalg
+import tensornetwork.linalg.initialization
+from tensornetwork import backends, backend_contextmanager
+from tensornetwork.tests import testing_utils
+
+#pylint: disable=no-member
+config.update("jax_enable_x64", True)
+
+
```
Comment (mganahl) [inline tensornetwork/linalg/tests/test_linalg.py:32]: Please add tests for `SymmetricBackend` where applicable

File: tensornetwork/__init__.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
     zeros
     )
 
+from tensornetwork.linalg.linalg import norm
```
[the remark this answers is not in the document]
Comment (alewis) [reply tensornetwork/__init__.py]: Added

File: tensornetwork/linalg/linalg.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  out = backend.qr(tensor.array, pivot_axis=pivot_axis,
+                   non_negative_diagonal=non_negative_diagonal)
+  Q, R = [Tensor(t, backend=backend) for t in out]
+  return Q, R
+
+
+def rq(
+    tensor: Tensor,
+    pivot_axis: int = -1,
+    non_negative_diagonal: bool = True
```
[the remark this answers is not in the document]
Comment (alewis) [reply tensornetwork/linalg/linalg.py]: Changed, good catch

File: tensornetwork/linalg/linalg.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+                    max_truncation_error=max_truncation_error,
+                    relative=relative)
+  tensors = [Tensor(t, backend=backend) for t in out]
+  return tuple(tensors)
+
+
+def qr(
+    tensor: Tensor,
+    pivot_axis: int = -1,
+    non_negative_diagonal: bool = True
```
[the remark this answers is not in the document]
Comment (alewis) [reply tensornetwork/linalg/linalg.py]: Changed

**Comment by alewis** [comment]
Testing the symmetric backend is going to be tricky for me because I have basically no idea how it works, what the data structure is, etc. 

Would it make sense to merge these, and then start a second PR that tests Symmetric by default?

**Comment by mganahl** [comment]
Sounds good, can you add an issue to remind us of doing that? thx!

**Comment by alewis** [comment]
Does anyone know why these linter errors have suddenly appeared?

**Comment by mganahl** [comment]
Fix conflicts and I'll merge

**Pull request merged**